app/models/data_update_script.rb

```diff
@@ -63,6 +63,10 @@ def file_class
     "#{self.class::NAMESPACE}::#{parsed_file_name.camelcase}".safe_constantize
   end
 
+  def save_error!(error)
+    update!(error: error)
+  end
+
```

Comment:
This might fail, I'm not entirely sure what Rails does here. If it calles `.to_s` on the exception object, it will only save the error message

As it usually really help to save the exception class as well and the error message itself might not be very self explanatory, I propose to replace `error: error` with `error: "#{e.class}:#{e.message}` or something similar

I also think this should be part of the `mark_as_failed!`, we probably don't need a separate method

Reply:
>As it usually really help to save the exception class as well and the error message itself might not be very self explanatory, I propose to replace error: error with error: "#{e.class}:#{e.message} or something similar

This is much better, thanks for the suggestion Rhymes:
error: "NoMethodError:undefined method `dosoemthing' for #<Article:0x00007fada0a37a78>">

>I also think this should be part of the mark_as_failed!, we probably don't need a separate method. 
This makes so much more sense, how did I miss it!